src/Nordea/Components/TextInput.elm

```diff
@@ -161,7 +176,28 @@ onEnterPress msg =
 
 view : List (Attribute msg) -> TextInput msg -> Html msg
 view attributes (TextInput config) =
-    if config.hasSearchIcon then
+    if config.hasRemoveInputButton then
+        Html.div
+            (css [ displayFlex, position relative ]
+                :: attributes
+            )
+            [ styled a
+                [ backgroundColor Colors.transparent
+                , position absolute
+                , right (rem 0.35)
+                , top (rem 0.35)
+                , border3 (rem 0.125) solid Colors.transparent
+                ]
+                (Maybe.values [ config.onRemoveInput |> Maybe.map onClick ])
+                [ Icons.roundedCross [ css [ width (rem 1.5), color Colors.mediumGray ] ] ]
```

Comment:
Do we need the wrapping a-tag, or can we put the onclick directly on the icon?
```suggestion
            [ Icons.roundedCross 
                [ Maybe.values [ config.onRemoveInput |> Maybe.map onClick ]
                , tabindex 0 -- remove if not needed. sometimes needed to make onclick work on divs.
                , css 
                    [ width (rem 1.5)
                    , color Colors.mediumGray
                    , position absolute
                    , right (rem 0.35)
                    , top (rem 0.35)
                    , padding (rem 1) -- should have some padding to increase clickable area for accessibility reasons. ideally, I'd say minimum 42x42px should be clickable
                    ]
                ]
```

Reply:
I had to leave the wrapping a-tag but added the padding.

Reply:
What was the problem? I think an a-tag may be wrong in regards to accessibility as it is not linking anywhere, so a button tag may be more appropriate in case a wrapper is needed